docs/documents/upsert.mdx

```diff
@@ -67,35 +65,204 @@ await client.collection("books").upsert([
+
+<CodeGroup>
 ```python Python
-client.collection("locations").upsert(
-[
-    {"_id": "point-1", "vector": [0.12, 0.67, 0.82, 0.53]},
-    {"_id": "point-2", "vector": [0.42, 0.53, 0.65, 0.33]},
-    {"_id": "point-3", "vector": [0.59, 0.33, 0.71, 0.61]},
-    {"_id": "point-4", "vector": [0.03, 0.88, 0.79, 0.41]},
-    {"_id": "point-5", "vector": [0.31, 0.64, 0.29, 0.59]},
-    {"_id": "point-6", "vector": [0.23, 0.14, 0.49, 0.91]},
-]
+
+from topk_sdk.schema import (
+    int,
+    text,
+    float,
+    bool,
+    f32Vector,
+    u8Vector,
+    binaryVector,
+    bytes,
+)
+
+client.collections().create(
+    "books",
+    schema={
+        "title": text(),
+        "published_year": int(),
+        "price": float(),
+        "is_published": bool(),
+        "float_embedding": f32Vector(dimension=1536).index(vector_index(metric="cosine")),
+        "u8_embedding": u8Vector(dimension=1536).index(vector_index(metric="euclidean")),
+        "binary_embedding": binaryVector(dimension=1536).index(vector_index(metric="hamming")),
+        "bytes": bytes(),
+    },
 )
 ```
 
-```js JavaScript
-// Reach out to us at support@topk.io if this is a blocker for you.
-// We will help you get started.
+```typescript Javascript
+import {
+  text,
+  int,
+  float,
+  bool,
+  f32Vector,
+  u8Vector,
+  binaryVector,
+  bytes,
+  vectorIndex,
+} from "topk-js/schema";
+
+await client.collections().create("books", {
+  title: text(),
+  published_year: int(),
+  price: float(),
+  is_published: bool(),
+  float_embedding: f32Vector({ dimension: 1536 }).index(vectorIndex({ metric: "cosine" })),
+  u8_embedding: u8Vector({ dimension: 1536 }).index(vectorIndex({ metric: "euclidean" })),
+  binary_embedding: binaryVector({ dimension: 1536 }).index(vectorIndex({ metric: "hamming" })),
+  bytes: bytes(),
+});
 ```
+</CodeGroup>
+
+Insert a document with all supported types:
 
+<CodeGroup>
+```python Python
+import f32_vector, u8_vector, binary_vector, bytes from topk_sdk.data;
+
+client.collection("books").upsert([
+  {
+    "_id": "1",
+    "title": "The Great Gatsby",
+    "published_year": 1925,
+    "price": 10.99,
+    "is_published": true,
+    "float_embedding": f32_vector([0.12, 0.67, 0.82, 0.53]),
+    "u8_embedding": u8_vector([0, 1, 2, 3]),
+    "binary_embedding": binary_vector([0, 1, 1, 0]),
+    "bytes": bytes([0, 1, 1, 0]),
+  },
+]);
+
+
+```
+
+```typescript Javascript
+import { f32Vector, u8Vector, binaryVector, bytes } from "topk-js/data";
+
+await client.collection("books").upsert([
+  {
+    _id: "1",
+    title: "The Great Gatsby",
+    published_year: 1925,
+    price: 10.99,
+    is_published: true,
+    float_embedding: [0.12, 0.67, 0.82, 0.53],
+    // or use the helper function:
+    // float_embedding: f32Vector([0.12, 0.67, 0.82, 0.53]),
+    u8_embedding: u8Vector([0, 1, 2, 3]),
+    binary_embedding: binaryVector([0, 1, 1, 0]),
+    bytes: bytes([0, 1, 1, 0]),
+  },
+]);
+```
+</CodeGroup>
+
+## Helper functions
```

Comment:
we should explain why these are needed to disambiguate between common types `number[]`